File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int lsum=0,rsum=0,n=cardPoints.length-1,max=0;
        for(int i=0;i<k;i++){
            lsum+=cardPoints[i];
        }
        max=lsum;
        for(int i=k-1;i>=0;i--){
            lsum-=cardPoints[i];
            rsum+=cardPoints[n--];
            max=Math.max(max,lsum+rsum);
        }
        return max;
    }
    
}

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        lsum,rsum=0,0
        for i in range(k):
            lsum+=cardPoints[i]
        
        maxSum=lsum
        rightIn=len(cardPoints)-1
        for j in range(k-1,-1,-1):
            lsum=lsum-cardPoints[j]
            rsum+=cardPoints[rightIn]
            rightIn-=1
            maxSum=max(maxSum,lsum+rsum)
        return maxSum
        